jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HTTP2ServerConnection.java

```diff
@@ -60,12 +62,14 @@ public class HTTP2ServerConnection extends HTTP2Connection implements Connection
     private static final Logger LOG = LoggerFactory.getLogger(HTTP2ServerConnection.class);
 
     private final HttpChannel.Factory httpChannelFactory = new HttpChannel.DefaultFactory();
+    private final Queue<HttpChannel> httpChannels = new ConcurrentLinkedQueue<>();
```

Comment:
This unbounded queue makes me uncomfortable. The TODO comment says recycling should use a `Pool` which does sound reasonable.

Reply:
I based this on previous implementation.  The unbounded queue will always be limited by the max number of concurrent streams per connection.  I thought about how could you attack this and make it create more, but I could not think of an approach.

I thought about putting a limit on the queue, but that's just another thing to configure.

I thought about using a pool, but ultimately our concurrent pool is just a queue anyway and we should not have too much concurrent access to the same connection.

Reply:
Right, there's a natural bound, external to this class, to the # of entries that can end up in this queue. It's worth commenting, though.